### ISSUE TITLE
[事前学習] - MoE

### ISSUE THREAD

**Issue by Taishi-N324** [llm-jp/pretrain-tasks#80]
# Overview

Fine-grained + shared expertsの効果を検証する 100Bスクラッチのバジェットで検証をする

1. top2, 8experts, 中間層サイズ=7168
   - [Baseline1である](https://github.com/llm-jp/experiments/issues/71)
2. top2, 16experts, 中間層サイズ=7168
   -  [GShard](https://arxiv.org/abs/2006.16668) の設定
3. top4, 16experts, 中間層サイズ=3584
   - Baseline1に対してのfine grained 
4. top8, 32experts, 中間層サイズ=1792
   - Baseline1に対してのfine grained 
5. パラメータをBaseline1と同じ
   -  top4, 32experts, 中間層サイズ=3584
   - 2に対してのfine grained 
6. top8, 64experts, 中間層サイズ=1792
   - 2に対してのfine grained
7. top3, 31experts + shared 1, 中間層サイズ=3584
   - 2に対してのfine grained + shared
8. top7, 63experts + shared 1, 中間層サイズ=1792
   - 2に対してのfine grained + shared
9. top1, 8experts, 中間層サイズ=7168
   - Switch Transformers
10. top4, 32experts, 中間層サイズ=1792
   - baeline9に対するfinrgrained
11. top8, 64experts, 中間層サイズ=896
   - OLMoE
12. top16, 128experts, 中間層サイズ=896
   - 2に対してのfine grained
13. top8, 128experts, 中間層サイズ=1792
   - ~2.9Bactivate/33Bfullパラメータ
14. top8, 64experts, 中間層サイズ=1536
   - 6に対して深くする 28layer 6は24layer
15. top8, 64experts, 中間層サイズ=1280
   - 6に対して深くする 32layer 6は24layer
16. top8, 64experts, 中間層サイズ=1792
   - 6に対して softmax topkのオーダを変える
17. top8, 64experts, 中間層サイズ=1792
   - 6で150kからrouter z loss 1e-3を追加
18. top8, 64experts, 中間層サイズ=1280
   - 15で150kからrouter z loss 1e-3を追加
19. top8, 64experts, 中間層サイズ=1792
   - 6でスクラッチでrouter z loss 1e-3を追加
20. top8, 64experts, 中間層サイズ=1792
   - 6でスクラッチでrouter z loss 1e-3とpresoftmaxの組み合わせ

 
 
# Details

[LLM-JP-MoE-2024年度後期実験計画](https://docs.google.com/document/d/1BWdaJpa-PFDf6UnvCW7HdYdlzOGFQNXX3xzc8CjJxIc/edit?usp=sharing) のexp3
Megatron-LMでの学習

# Resources

* **計算機**
  * クラスタ: Sakura (Ishikari)
  * ノード種別: gpu-small (H100x8)
  * ノード台数: 状況に応じて
* **コード**
  * リポジトリ: https://github.com/llm-jp/Megatron-LM/tree/1250d2cbeb8686171a5ab026d1fc40a25b625bc5
  * コミット: 1250d2cbeb8686171a5ab026d1fc40a25b625bc5
* **入力データ**:
  * LLM-jp v3.1 コーパス: `sakura:/data/llm-jp-corpus/v3.{0,1}.0`
* **出力データ**:
  * 保存先:
    * `sakura:/data/experiments/0080
    * `sakura:/home/shared/experiments/0080
  * データ内訳:
    * checkpoint: **FIXME** TB （バッファ容量を含む）
* **W&B ログ**:
  * https://wandb.ai/llm-jp/v3-8x1.8b
* **開始日**: 2024-11-DD
* **終了予定日**: 2024-MM-DD （バッファ期間を含む）
